src/app/donation-start/donation-start-form/donation-start-form-new.component.html

```diff
@@ -29,7 +29,19 @@
 
         <div class="donation-input donation-input-main">
             <biggive-text-input currency="{{campaign.currencyCode}}">
-              <label slot="label" for="donationAmount">Enter your donation amount</label>
+              <!-- If a charity name is more than about 20 chars it may not fit in the label space and would
+                   make the input unusable
+               -->
+              <label
+                *ngIf="campaign.charity.name.length < 20; else donationLabelTruncated"
+                slot="label"
+                for="donationAmount"
+              >
+                Donation to {{campaign.charity.name}}
+              </label>
+              <ng-template #donationLabelTruncated>
+              <label slot="label" for="donationAmount">Donation to {{campaign.charity.name.slice(0, 17)}}…</label>
```

Comment:
this is not perfect because slice splits by code units, would be better to split by graphemes. Maybe using Intl.Segmenter is the proper way, but the chance we have a very long charity name with more than one code unit per grapheme (e.g. a letter combined with a diacritic, or an emoji with a skin tone applied) is pretty low.